pkg/custompluginmonitor/types/config.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2017 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"fmt"
+	"time"
+
+	"os"
+
+	"k8s.io/node-problem-detector/pkg/types"
+)
+
+var (
+	defaultGlobalTimeout        = 5 * time.Second
+	defaultGlobalTimeoutString  = defaultGlobalTimeout.String()
+	defaultInvokeInterval       = 30 * time.Second
+	defaultInvokeIntervalString = defaultInvokeInterval.String()
+	defaultMaxOutputLength      = 80
+	defaultConcurrency          = 3
+
+	customPluginName = "custom"
+)
+
+type pluginGlobalConfig struct {
+	// InvokeIntervalString is the interval string at which plugins will be invoked.
+	InvokeIntervalString *string `json:"invoke_interval, omitempty"`
+	// TimeoutString is the global plugin execution timeout string.
+	TimeoutString *string `json:"timeout, omitempty"`
+	// InvokeInterval is the interval at which plugins will be invoked.
+	InvokeInterval *time.Duration `json:"-"`
+	// Timeout is the global plugin execution timeout.
+	Timeout *time.Duration `json:"-"`
+	// MaxOutputLength is the maximum plugin output message length.
+	MaxOutputLength *int `json:"max_output_length, omitempty"`
+	// Concurrency is the number of concurrent running plugins.
+	Concurrency *int `json:"concurrency, omitempty"`
+}
+
+// Custom plugin config is the configuration of custom plugin monitor.
+type CustomPluginConfig struct {
+	// Plugin is the name of plugin which is currently used.
+	// Currently supported: custom.
+	Plugin string `json:"plugin, omitempty"`
+	// PluginConfig is global plugin configuration.
+	PluginGlobalConfig pluginGlobalConfig `json:"pluginConfig, omitempty"`
+	// Source is the source name of the custom plugin monitor
+	Source string `json:"source"`
```

Comment:
Probably, just have `Plugin` name, and use `Plugin` name as the source?

Reply:
This is mainly used to let users define the user specified event source. This should be retained, IMO.